### DISCUSSION TITLE
Can't run run.bat file

### DISCUSSION THREAD

**Discussion by arjunkaushal123** [snipem/gt7dashboard#13] [Q&A]
can't run the run.bat file, can you please let me know the command (already replaced my playstation's ip address inside the file)

![Screenshot (64)](https://github.com/snipem/gt7dashboard/assets/74197122/0ca5a0a9-9c19-4731-a23a-ea62d76d86a8)

**Comment by snipem** [comment]
Can you run `run.bat` from `cmd.exe` and post the output?

**Reply by arjunkaushal123** [reply]
![image](https://github.com/snipem/gt7dashboard/assets/74197122/3fd1fcf6-d358-47a5-acb5-1f70c3118f19)

Here

**Reply by snipem** [reply]
I just found out that Windows 11 defaults to powershell which is not compatible to classic batch files. Download, edit and run this file:  https://github.com/snipem/gt7dashboard/blob/improvements/run.ps1 It is in a feature branch. I will update in the future.

**Reply by arjunkaushal123** [reply]
![image](https://github.com/snipem/gt7dashboard/assets/74197122/a5e4d865-bac6-449f-9e46-3592b682766e)

didn't work

**Reply by snipem** [reply]
Try .\run.ps1

**Reply by snipem** [reply]
Please post text instead of screenshots of error messages

**Reply by arjunkaushal123** [reply]
![image](https://github.com/snipem/gt7dashboard/assets/74197122/750b6e91-f78e-460b-9a9a-766e9b50ff12)

**Reply by arjunkaushal123** [reply]
my apologies,
The output:
"
'.\run.ps1' is not recognized as an internal or external command,
operable program or batch file.
"

**Reply by snipem** [reply]
Try typing in every command from the file line by line. I cannot guess what command is failing in your setup. Please post the results here.

**Reply by arjunkaushal123** [reply]
1
![Screenshot (80)](https://github.com/snipem/gt7dashboard/assets/74197122/9dc5fedd-f096-4d1f-8e4e-a01097dc7f5d)

2
![Screenshot (81)](https://github.com/snipem/gt7dashboard/assets/74197122/581a53a4-9b38-429d-924f-d07cbfcfbbb6)

3
![Screenshot (82)](https://github.com/snipem/gt7dashboard/assets/74197122/097eec7f-75c9-42cd-8964-a9450bed505b)

4
![Screenshot (83)](https://github.com/snipem/gt7dashboard/assets/74197122/15c3d1be-15f4-43b7-a0d6-9ec09897991f)

**Reply by snipem** [reply]
No, please type the content from `run.ps1` or `run.bat` line by line into the terminal. This should give a hint about what command is wrong.

You should also try Googling how to run powershell scripts (ps1) and Windows batch files (bat) from either Powershell or cmd.exe. This would help in debugging.

**Reply by arjunkaushal123** [reply]
1

![Screenshot (84)](https://github.com/snipem/gt7dashboard/assets/74197122/1350f32e-318f-4a09-a2be-78f69568ebdd)


2

![Screenshot (85)](https://github.com/snipem/gt7dashboard/assets/74197122/81dd39ed-8f6f-4cf9-a8a7-a30e4d1f0d20)


3

![Screenshot (86)](https://github.com/snipem/gt7dashboard/assets/74197122/8b6e3058-c85e-40b5-8e1c-3a0f6bb6825a)


4

![Screenshot (87)](https://github.com/snipem/gt7dashboard/assets/74197122/6f3b1482-688a-4cd9-ab3f-5cea71b50e74)


5

![Screenshot (88)](https://github.com/snipem/gt7dashboard/assets/74197122/3c10a340-de93-4b3d-9e92-f5350a8a8cfb)


6

![Screenshot (89)](https://github.com/snipem/gt7dashboard/assets/74197122/7a5148ea-f57c-44c3-8d2c-e91c66a54ba6)


sure, ill revert with that shortly

**Reply by snipem** [reply]
The cause is in screenshot 5 and also with the solution: `error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/`

Over at gtplanet.com someone had the same problem: https://www.gtplanet.net/forum/threads/overview-of-gt7-telemetry-software.418011/post-14030492 See how they solved it.

**Reply by arjunkaushal123** [reply]
Realised my fault..
downloaded that and am able to run run.ps1 through powershell, but now my screen's blank when i open the bokeh application through google chrome

![Screenshot (98)](https://github.com/snipem/gt7dashboard/assets/74197122/89a09b78-d78e-42ee-be68-cd966dcbf940)

It's prompting (Exception: No IP set in env var GT7_PLAYSTATION_IP).
rechecked all the applications where i had to edit my playstation's IP, but ive filled out everyplace for it.

![image](https://github.com/snipem/gt7dashboard/assets/74197122/32d0bfb4-ca7d-400c-ab53-9a57c1dfc634)

suspect this to be concerning too

**Reply by snipem** [reply]
You have to run all the commands from the batch and powershell files in succession otherwise it will not run and the application will not know your IP address. This is in your second to last screenshot:

```No IP set in env var GT7_PLAYSTATION_IP```

It would be of great help if you could get the bat or powershell file to run.

**Reply by arjunkaushal123** [reply]
I can run the powershell file but the bokeh app is not displaying anything on any browser.
still facing the same issue with run.bat file.

![Screenshot (102)](https://github.com/snipem/gt7dashboard/assets/74197122/d3a751a4-ec65-41dd-86e8-ee1954566448)


I've tried to approach this in different ways but all have seemed to fail to give me the same prompt back
```
'run.bat' is not recognized as an internal or external command,
operable program or batch file.
```

**Comment by snipem** [comment]
Maybe this is a formatting error of the file. I converted it to Windows line endings with unix2dos. I have no chance on testing it myself. Since I do not own Windows. Can you open the file and type in the contents line by line? I would be happy if you can send me the fixed version.

**Comment by IRIX64** [comment]
![grafik](https://github.com/snipem/gt7dashboard/assets/122752155/157e63f4-6a5d-4599-9d62-783106473cca)

That syntax is completly wrong. Under "dos"  you use   "set name=value" to specify a variable. You havent execute the right one i think.

I download the package a few minutes ago and the run.bat  (it have windows line breaks) work fine if you modify line #3 in a proper way.  You can also run the commands one by one on the command line if you copy and paste it from a open notepad.

Regards
Joerg

**Reply by snipem** [reply]
Hi Jörg. Could you tell me the right syntax for both batch and powershell? I will gladly update it.

**Comment by IRIX64** [comment]
You havent make something wrong... the run.bat works and doesnt report any error as long as someone replace the <> also and not simply add the IP between.

I dont see any powershell in your project?  Adding a posh start script isnt necessary and in my opinion your run.bat is fine.

Regards,
Joerg

**Reply by snipem** [reply]
I created one in a feature branch because of this issue: https://github.com/snipem/gt7dashboard/blob/improvements/run.ps1

I expected Windows 11 to default to Powershell when opening batch files because of this issue. I do not use Windows myself, so I am not able to check myself. On the other hand a prototypical pipeline run with a GitHub Windows Runner did open the powershell file successfully.

**Reply by arjunkaushal123** [reply]
ran powershell file again, got it without the error/exception.
![image](https://github.com/snipem/gt7dashboard/assets/74197122/936c0d13-de18-44d4-ad01-105d46ed8f23)

would it work if i switched to my playstation or still would need to run run.bat successively because i double click, run with administrator, do " .\run.bat " in cmd, it opens something briefly and closes it.

**Reply by arjunkaushal123** [reply]
![image](https://github.com/snipem/gt7dashboard/assets/74197122/d770dbf7-feee-4419-9151-8bc6c268f13e)
played a bit of qualifying for daily race c after booting bookeh app. still no display but its showing that it has been loading content for the past 20mins

**Reply by snipem** [reply]
It should display at least an empty UI with a red light if there is no connection. Does the log say something?

**Reply by arjunkaushal123** [reply]
nothing direct. Again, bokeh app is blank while the connection's been established.

pinged ps's ip too after some guy from gtplanet forum asked me just to crosscheck, 32 bytes went through with 4ping
(https://www.gtplanet.net/forum/threads/overview-of-gt7-telemetry-software.418011/post-14043534)

**Reply by arjunkaushal123** [reply]
![image](https://github.com/snipem/gt7dashboard/assets/74197122/05eb5751-a661-48a1-aceb-0bc419d4a3cb)
one of the issues popping up when I scroll through developer tools

**Reply by snipem** [reply]
You have at least one bokeh instance running (error message). Stop it and start again.

**Reply by arjunkaushal123** [reply]
apologies but how do i do that?

**Reply by snipem** [reply]
Windows restart is sufficient

**Reply by arjunkaushal123** [reply]
![image](https://github.com/snipem/gt7dashboard/assets/74197122/57fd5dfb-f608-46c6-990a-343fed9d0a41)
No luck mate

**Reply by snipem** [reply]
I think from this point on you would be better off if you would run the Docker container on a capable machine (some routers support Docker images for example).

As others reported the Windows batch files work. But I have to say it is not a trivial process to set up the whole environment.

**Comment by IRIX64** [comment]
Post #1
The explorer hide the file extensions but displays "run.bat" which indicates that you modify the entire filename to something different.

In later steps you try to start the bokeh by your self but there is a '.' missing at the end and thats why a path error is returned.  The documentation on the project page gives me everything to get it working under windows so just follow it ;)
